tests/test_client_functional.py

```diff
@@ -2202,6 +2202,24 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "status", [200, 201, 301, 400, 403, 500], ids=lambda x: f"status_code: {x}"
+)
+async def test_ok_from_status(aiohttp_client, status) -> None:
+
+    async def handler(request):
+        return web.Response(status=status, body=b'')
+
+    app = web.Application()
+    app.router.add_route('GET', '/endpoint', handler)
+    client = await aiohttp_client(app, raise_for_status=False)
+    resp = await client.get('/endpoint')
+
+    if resp.status < 400:
+        assert resp.ok
+    else:
+        assert not resp.ok
```

Comment:
You shouldn't add such logic into tests. Either factor it out into the params matrix or split the test into two — positive and negative.

Reply:
```suggestion
    assert resp.ok is expected_ok
```